tests/UserControllerTest.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+use App\Models\User;
+
+class UserControllerTest extends TestCase {
+    
+    /**
+     * @dataProvider providerListAction
+     */
+    public function testForListAction($input) {
+        $service = $this->getMockBuilder("App\Services\UserService")
+                ->setConstructorArgs(['ut'])
+                ->getMock();
+        $userController = new App\Controllers\UserController();
+        $userController->setService($service);
+        
+        $service->expects($this->once())     
+                ->method("getAllUsers")
+                ->will($this->returnValue(new User()));
+        
+        $userController->listAction($service->getAllUsers($input));
```

Comment:
remove $service->getAllUsers($input)
it should looks like this: $userController->listAction()